test/test_python_symint.py

```diff
@@ -0,0 +1,84 @@
+from typing import Iterator, List
+import torch
+from enum import Enum
+import torch
+#from base_tensor import BaseTensor
+from torch.testing._internal.common_utils import run_tests, TestCase
+from torch.utils._pytree import tree_map
+import contextlib
+
+@contextlib.contextmanager
+def no_dispatch() -> Iterator[None]:
+    guard = torch._C._DisableTorchDispatch()  # type: ignore[attr-defined]
+    try:
+        yield
+    finally:
+        del guard
+
+class Op(Enum):
+    SIZE = 1
+    ADD = 2
+
+class AOTAutogradSymInt:
+    def __init__(self, op: Op, children: List["AOTAutogradSymInt"]):
+        self._op = op
+        self._children = children
+
+    def __str__(self):
+        return f"{self._op}"
+
+    def __add__(self, other):
+        return AOTAutogradSymInt(Op.ADD, [self, other])
```

Comment:
weren't you going to have an inner and outer? Shouldn't this call `new_symint`?

Reply:
`AOTAutogradSymInt` is the inner object. We are wrapping it into a `SymIntNode` below: `sym1 = torch._C.SymbolicIntNode.new_symint(AOTAutogradSymInt(Op.SIZE, [a, 0]))`